File: DeepSpaceJava/src/ShieldBooster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author Samuel Medina
 * @author Laura Sánchez
 */

class ShieldBooster {
    
    // Definición de atributos
    private String name;
    private float boost;
    private int uses;
    
    // Constructor 
    public ShieldBooster(String n,float b,int u){
        name = n;
        boost = b;
        uses = u;
    }
    
    // Constructor de copias
    public ShieldBooster(ShieldBooster s){
        ShieldBooster shield = new ShieldBooster(s.name, s.boost, s.uses);
    }
    
    // Consultores
    public String getName(){
        return name;
    }
    
    public float getBoost(){
        return boost;
    }
    
    public int getUses(){
        return uses;
    }
    
    /* Useit: Decrementa uses si uses>0 y
    Devuelve el valor de boost, duelve 1.0 en otro caso*/
    public float useIt(){
        if(uses>0){
            uses--;
            return boost;
        }else
            return (float) 1.0;
    }
    
    
    }
   
   
    
    

File: DeepSpaceRuby/lib/shield_booster.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module DeepSpace
  class ShieldBooster
    def initialize(n,b,u)
      @name = n
      @boost = b
      @uses = u

    end

    def newCopy(s)
      copy =ShieldBooster.new(s.name, s.boost, s.uses)
      return copy
    end

    attr_reader:name

    attr_reader:boost

    attr_reader:uses

    def useIt
      if @uses > 0
        @uses=@uses-1
        return @boost
      else
        return 1.0
      end
    end
  end
end